mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -744,7 +744,13 @@ private[classification] class MultiClassSummarizer extends Serializable {
   def countInvalid: Long = totalInvalidCnt
 
   /** @return The number of distinct labels in the input dataset. */
-  def numClasses: Int = distinctMap.keySet.max + 1
+  def numClasses: Int = try {
+    distinctMap.keySet.max + 1
+  } catch {
+    case e: UnsupportedOperationException =>
```

Comment:
@srowen I made changes to return 0 when the hashtable is empty. 
I found that 
301       if (numInvalid != 0) {
302        val msg = s"Classification labels should be in {0 to ${numClasses - 1} " +
303          s"Found $numInvalid invalid labels."
304        logError(msg)
305        throw new SparkException(msg)
306      }
So, for the empty case, it will throw an exception anyway after train. Then, returning 0 or throw an exception won't make much difference.